EIPS/eip-7702.md

```diff
@@ -41,18 +41,15 @@ We introduce a new [EIP-2718](./eip-2718.md) transaction, "set code transaction"
 ```
 rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, value, data, access_list, authorization_list, signature_y_parity, signature_r, signature_s])
 
-authorization_list = [[chain_id, address, nonce, y_parity, r, s], ...]
+authorization_list = [[unchained, address, nonce, y_parity, r, s], ...]
 ```
 
-The fields `chain_id`, `nonce`, `max_priority_fee_per_gas`, `max_fee_per_gas`, `gas_limit`, `destination`, `value`, `data`, and `access_list` of the outer transaction follow the same semantics as [EIP-4844](./eip-4844.md). *Note, this means a null destination is not valid.*
+The transaction fields `chain_id`, `nonce`, `max_priority_fee_per_gas`, `max_fee_per_gas`, `gas_limit`, `destination`, `value`, `data`, and `access_list` follow the same semantics as [EIP-4844](./eip-4844.md). *Note, this means a null destination is not valid.*
 
-The `authorization_list` is a list of tuples that store the address to code which the signer desires to execute in the context of their EOA. The transaction is considered invalid if the length of `authorization_list` is zero.
-
-The transaction is also considered invalid when any field in an authorization
-tuple cannot fit within the following bounds:
+`authorization_list` is a list of tuples that store the address to code which the signer desires to execute in the context of their EOA. The transaction is considered invalid if the length of `authorization_list` is zero. It is also considered invalid when any field in an authorization tuple exceeds the following bounds:
 
 ```python
-assert auth.chain_id < 2**64
+assert auth.unchained == 0 || auth.unchained == 1
```

Comment:
This specific check looks like the `yParity` check (where `yParity` is also either 0 or 1). 
```suggestion
assert auth.unchained < 2**8
```

There should be an added check for each tuple to validate if the `auth.unchained` is 0 or 1 (if not, skip the tuple).

Reply:
Yes, I think this line should be reverted. This section has to do with decoding txs and I think 1 byte is the highest resolution we want here. I think many languages represent boolean as 1 byte value despite it only needing 1 bit of information.

We would verify `unchained` is 0 or 1 in the tuple validation.